nx/source/services/capsa.c

```diff
@@ -0,0 +1,160 @@
+#define NX_SERVICE_ASSUME_NON_DOMAIN
+#include <string.h>
+#include <time.h>
+#include "service_guard.h"
+#include "runtime/hosversion.h"
+#include "services/applet.h"
+#include "services/capsa.h"
+
+static Service g_capsaSrv;
+
+NX_GENERATE_SERVICE_GUARD(capsa);
+
+Result _capsaInitialize(void) {
+    return smGetService(&g_capsaSrv, "caps:a");
+}
+
+void _capsaCleanup(void) {
+    serviceClose(&g_capsaSrv);
+}
+
+Service* capsaGetServiceSession(void) {
+    return &g_capsaSrv;
+}
+
+Result capsaGetAlbumFileCount(CapsAlbumStorage storage, u64* count) {
+    return serviceDispatchInOut(&g_capsaSrv, 0, storage, *count);
```

Comment:
In IPC CapsAlbumStorage is an u8, so store it in a tmp u8 field (likewise for everything else using it).